apps/arm/main.cpp

```diff
@@ -180,7 +181,85 @@ static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-    return MBED_SUCCESS; // TODO
+    /** ---------------------------------------------------------
+     * | 24 bits  | 32 bits  | 3 bits   | 1 bit   |  4 bits      |
+     * | unused   | data     | param    | vel/pos |  Actuator ID |
+     *  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    */
+    // Prepare bitfield to break down CAN data
+    struct{
```

Comment:
I think this is bad style to be defining structs within functions.

Reply:
its definetly odd. i feel like this time it's justified because it is not used anywhere but within the function. i've thought about it some and think this might be justified in this case. if you feel strongly that it's bad i can change it.

Reply:
it would be useful to have this in a header file that could be reused in ROS code. We could have a folder of these, one file per board?  
This could be done in a followup issue.

Additionally, it might be a good idea to start implementing a version number within with our board-to-ros communication msgs (and accompanying logic) so we can tell if there's a version mismatch right away. Another idea could be to wrap our can txns in protobuf, but i think that may be a little to heavy for our use

@jetkov thoughts?

Reply:
@wmmc88 i think ensuring the versions are the same is important. I was thinking it could be implemented as some warning that CI raises when a PR that causes a mismatch is made. is this plausible/easier than implementing version numbers?